jsontests/src/lib.rs

```diff
@@ -459,7 +465,6 @@ impl ExitBehavior {
 		if self.save_current_step {
 			listener.save_current_step();
 		} else if listener.current_step_consumed
-			&& listener.current_step.depth <= 1
```

Comment:
There were some other tests failing without this, but I think with the above change that should be fixed.